lib/cmd/fh3/app/stage.js

```diff
@@ -1,60 +1,160 @@
-
-base.preCmd = function(argv, cb) {
-  var data = {
-    env: argv.env,
-    app: argv.app,
-    gitRef: argv.gitRef || {}
-  };
-
-  // Default to master like studio
-  if (!argv.gitRef) {
-    data.gitRef.type='branch';
-    data.gitRef.value='master';
-  }
 
-  if (argv.runtime) {
-    data.runtime = argv.runtime;
-  }
-  if (argv.clean) {
-    data.clean = argv.clean === 'true';
-  }
-  if (argv.autodeploy) {
-    data.autoDeploy = argv.autodeploy !== 'false';
+module.exports = {
+  'desc' : i18n._('Stages a cloud application.'),
+  'examples' :
+  [{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment>',
+    desc : i18n._("Deploy of cloud application <app-id> on <environment> (Defaults to branch:master)")
+  },{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment> --runtime=<runtime> --clean=false --autodeploy=true',
+    desc : i18n._("Deploy of cloud application <cloud-app-id> on <environment> with the <runtime> without Clean Stage option selected and 'Auto Deploy' option selected.")
+  },{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment> --runtime=<runtime> --gitRefType=<gitRef-type> --gitRefHash=<gitRef-hash>',
+    desc : i18n._("Deploy of cloud application <cloud-app-id> on <environment> with the <runtime> with 'branch' or 'tag' informed and the commit hash that you'd like to deploy.")
+  },{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment> --runtime=<runtime> --gitRefValue=<gitRef-value>',
+    desc : i18n._("Deploy of cloud application <cloud-app-id> on <environment> with the <runtime> with the name of the branch that you'd like to deploy.")
+  }],
+  'demand' : ['app', 'env'],
+  'method' : 'post',
+  'alias' : {
+    'app' : 'a',
+    'env' : 'e',
+    'runtime' : 'r',
+    'clean' : 'c',
+    'autodeploy': 'ad',
+    'gitRefType' : 'gt',
+    'gitRefHash' : 'gh',
+    'gitRefValue' : 'gv',
+    0 : 'app',
+    1 : 'env'
+  },
+  'describe' : {
+    'app' : i18n._('Unique 24 character GUID of the cloud app you want to see resource info for'),
+    'env' : i18n._('The cloud environment your app is running in'),
+    'runtime' : i18n._("The Node.js runtime of your application. (E.g. node08, node010 or node4.). To check the available runtimes for this app use '$fhc app runtimes --id=<cloud-app-id>'"),
+    'clean' : i18n._("Options 'true' or 'false'. Do a full, clean stage. Cleans out all old application log files, removes cached node modules and does an 'npm install' from scratch"),
+    'autodeploy' : i18n._("Options 'true' or 'false'. Deploy the app automatically when the app repo is updated"),
+    'gitRefType' : i18n._("Specifies if you'd like to deploy a 'branch' or 'tag'"),
+    'gitRefHash' : i18n._("The commit hash you'd like to deploy if gitRef.type is 'branch'. The default value is HEAD."),
+    'gitRefValue' : i18n._("The name of the branch you'd like to deploy")
+  },
+  'preCmd' : function(params, cb) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Function has a complexity of 25 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)